module/friendship/app/query/list_friends.go

```diff
@@ -0,0 +1,75 @@
+package query
+
+import (
+	"context"
+
+	"github.com/phantranhieunhan/s3-assignment/common"
+	"github.com/phantranhieunhan/s3-assignment/common/logger"
+	"github.com/phantranhieunhan/s3-assignment/module/friendship/domain"
+)
+
+type ListFriendsRepo interface {
+	GetFriendshipByUserIDAndStatus(ctx context.Context, userID string, status ...domain.FriendshipStatus) (domain.Friendships, error)
+}
+
+type UserRepo interface {
+	GetUserIDsByEmails(ctx context.Context, emails []string) (map[string]string, error)
+	GetEmailsByUserIDs(ctx context.Context, userIDs []string) (map[string]string, error)
+}
+
+type ListFriendsHandler struct {
+	repo     ListFriendsRepo
+	userRepo UserRepo
+}
+
+func NewListFriendsHandler(repo ListFriendsRepo, userRepo UserRepo) ListFriendsHandler {
+	return ListFriendsHandler{
+		repo:     repo,
+		userRepo: userRepo,
+	}
+}
+
+func (h ListFriendsHandler) Handle(ctx context.Context, email string) ([]string, error) {
+	var zList []string
```

Comment:
Review naming convention! What does the `z` represents in the name of this variable ?